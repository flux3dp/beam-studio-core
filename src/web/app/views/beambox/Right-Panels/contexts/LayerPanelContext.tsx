import * as React from 'react';
import { ILayerPanelContext } from 'interfaces/IContext';

export const LayerPanelContext = React.createContext<ILayerPanelContext>(null);

interface Props {
  children: React.ReactNode;
}

export class LayerPanelContextProvider extends React.Component<Props> {
  private contextValue = null;

  private selectedLayers: string[] = [];

  constructor(props) {
    super(props);
    this.updateContextValue();
  }

  updateContextValue = (): void => {
    const {
      selectedLayers,
      setSelectedLayers,
      updateLayerPanel,
    } = this;
    this.contextValue = {
      selectedLayers,
      setSelectedLayers,
      updateLayerPanel,
    };
  };

  setSelectedLayers = (selectedLayers: string[]): null => {
    this.selectedLayers = [...selectedLayers];
    this.updateContextValue();
    this.forceUpdate();
    return null;
  };

  updateLayerPanel = (): void => {
    this.forceUpdate();
  };

  render() {
    const { children } = this.props;
    return (
      <LayerPanelContext.Provider value={this.contextValue}>
        {children}
      </LayerPanelContext.Provider>
    );
  }
}
