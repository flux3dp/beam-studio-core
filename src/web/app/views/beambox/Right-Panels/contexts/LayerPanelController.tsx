import eventEmitterFactory from 'helpers/eventEmitterFactory';

const layerPanelEventEmitter = eventEmitterFactory.createEventEmitter('layer-panel');

const updateLayerPanel = (): void => {
  layerPanelEventEmitter.emit('UPDATE_LAYER_PANEL');
};

const getSelectedLayers = (): string[] => {
  const response = {
    selectedLayers: [],
  };
  layerPanelEventEmitter.emit('GET_SELECTED_LAYERS', response);
  return response.selectedLayers;
};

const setSelectedLayers = (selectedLayers: string[]): void => {
  layerPanelEventEmitter.emit('SET_SELECTED_LAYERS', selectedLayers);
};

export default {
  updateLayerPanel,
  getSelectedLayers,
  setSelectedLayers,
};
