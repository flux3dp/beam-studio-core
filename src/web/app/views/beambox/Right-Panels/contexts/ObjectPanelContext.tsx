import React from 'react';

import eventEmitterFactory from 'helpers/eventEmitterFactory';

interface IObjectPanelContext {
  dimensionValues: any;
  updateDimensionValues: (newValues: any) => void;
  getDimensionValues: (response: {
    dimensionValues: any,
  }, key?: string,) => any;
  updateObjectPanel: () => void;
}

export const ObjectPanelContext = React.createContext<IObjectPanelContext>({
  dimensionValues: {},
  updateDimensionValues: () => { },
  getDimensionValues: () => { },
  updateObjectPanel: () => { },
});
const objectPanelEventEmitter = eventEmitterFactory.createEventEmitter('object-panel');
const minRenderInterval = 200;

interface State {
  lastUpdateTime: number;
}

export class ObjectPanelContextProvider extends React.Component<any, State> {
  private dimensionValues: any;

  private updateTimeout: NodeJS.Timeout;

  constructor(props) {
    super(props);
    this.dimensionValues = {};
    this.state = {
      lastUpdateTime: Date.now(),
    };
  }

  componentDidMount() {
    objectPanelEventEmitter.on('UPDATE_DIMENSION_VALUES', this.updateDimensionValues.bind(this));
    objectPanelEventEmitter.on('GET_DIMENSION_VALUES', this.getDimensionValues.bind(this));
    objectPanelEventEmitter.on('UPDATE_OBJECT_PANEL', this.updateObjectPanel.bind(this));
  }

  componentWillUnmount() {
    objectPanelEventEmitter.removeAllListeners();
  }

  updateDimensionValues = (newValues: any): void => {
    this.dimensionValues = {
      ...this.dimensionValues,
      ...newValues,
    };
  };

  getDimensionValues = (response: {
    dimensionValues: any,
  }, key?: string): void => {
    response.dimensionValues = key ? this.dimensionValues[key] : this.dimensionValues;
  };

  updateObjectPanel = (): void => {
    clearTimeout(this.updateTimeout);
    const time = Date.now();
    const { lastUpdateTime } = this.state;
    if (time - lastUpdateTime >= minRenderInterval) {
      this.setState({ lastUpdateTime: time });
    } else {
      this.updateTimeout = setTimeout(() => {
        this.setState({
          lastUpdateTime: lastUpdateTime + minRenderInterval,
        });
      }, lastUpdateTime + minRenderInterval - time);
    }
  };

  render(): JSX.Element {
    const { children } = this.props;
    const {
      dimensionValues,
      updateDimensionValues,
      getDimensionValues,
      updateObjectPanel,
    } = this;
    return (
      <ObjectPanelContext.Provider value={{
        dimensionValues,
        updateDimensionValues,
        getDimensionValues,
        updateObjectPanel,
      }}
      >
        {children}
      </ObjectPanelContext.Provider>
    );
  }
}
