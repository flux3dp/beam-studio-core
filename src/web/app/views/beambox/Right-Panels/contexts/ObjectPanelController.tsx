import eventEmitterFactory from 'helpers/eventEmitterFactory';

const objectPanelEventEmitter = eventEmitterFactory.createEventEmitter('object-panel');

const updateDimensionValues = (newValue: any): void => {
  objectPanelEventEmitter.emit('UPDATE_DIMENSION_VALUES', newValue);
};

const getDimensionValues = (key: string): any => {
  const response = {
    dimensionValues: {},
  };
  objectPanelEventEmitter.emit('GET_DIMENSION_VALUES', key, response);
  return response.dimensionValues;
};

const updateObjectPanel = (): void => {
  objectPanelEventEmitter.emit('UPDATE_OBJECT_PANEL');
};

export default {
  updateObjectPanel,
  updateDimensionValues,
  getDimensionValues,
};
