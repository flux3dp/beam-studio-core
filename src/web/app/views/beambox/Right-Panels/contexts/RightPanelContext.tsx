import * as React from 'react';

export const RightPanelContext = React.createContext(null);

interface State {
  mode: string;
  selectedElement: Element;
}

export class RightPanelContextProvider extends React.Component<any, State> {
    constructor(props) {
        super(props);
        this.state = {
            mode: 'element',
            selectedElement: null,
        }
    }

    setMode = (mode) => {
        const {mode: currentMode} = this.state;
        if (['path-edit'].includes(mode) || currentMode !== mode) {
            this.setState({ mode });
        }
    }

    setSelectedElement = (elems) => {
        if (elems !== this.state.selectedElement) {
            (document.activeElement as HTMLInputElement).blur();
        }
        this.setState({selectedElement: elems});
    }

    render() {
        const {
            setMode,
            setSelectedElement,
        } = this;
        const {
            mode,
            selectedElement,
        } = this.state;
        return (
            <RightPanelContext.Provider value={{
                setMode,
                mode,
                setSelectedElement,
                selectedElement,
            }}>
                {this.props.children}
            </RightPanelContext.Provider>
        );
    }
};
