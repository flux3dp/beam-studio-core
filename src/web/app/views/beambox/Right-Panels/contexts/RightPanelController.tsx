import BeamboxGlobalInteraction from 'app/actions/beambox/beambox-global-interaction';
import eventEmitterFactory from 'helpers/eventEmitterFactory';

const rightPanelEventEmitter = eventEmitterFactory.createEventEmitter('right-panel');

const setSelectedElement = (elem): void => {
  if (!elem) {
    BeamboxGlobalInteraction.onObjectBlur();
  } else {
    BeamboxGlobalInteraction.onObjectBlur();
    BeamboxGlobalInteraction.onObjectFocus([elem]);
  }
  rightPanelEventEmitter.emit('SET_SELECTED_ELEMENT', elem);
};

const toPathEditMode = (): void => {
  rightPanelEventEmitter.emit('SET_MODE', 'path-edit');
};

const toElementMode = (): void => {
  rightPanelEventEmitter.emit('SET_MODE', 'element');
};

export default {
  setSelectedElement,
  toPathEditMode,
  toElementMode,
};
