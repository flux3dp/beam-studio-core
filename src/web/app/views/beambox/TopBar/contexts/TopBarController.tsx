import eventEmitterFactory from 'helpers/eventEmitterFactory';

const topBarEventEmitter = eventEmitterFactory.createEventEmitter('top-bar');

const updateTopBar = (): void => {
  topBarEventEmitter.emit('UPDATE_TOP_BAR');
};

const setElement = (elem: Element | null): void => {
  topBarEventEmitter.emit('SET_ELEMENT', elem);
};

const setFileName = (fileName: string): void => {
  topBarEventEmitter.emit('SET_FILE_NAME', fileName);
};

const setHasUnsavedChange = (hasUnsavedChange: boolean): void => {
  topBarEventEmitter.emit('SET_HAS_UNSAVED_CHANGE', hasUnsavedChange);
};

const getTopBarPreviewMode = (): boolean => {
  const response = {
    isPreviewMode: false,
  };
  topBarEventEmitter.emit('GET_TOP_BAR_PREVIEW_MODE', response);
  return response.isPreviewMode;
};

const setShouldStartPreviewController = (shouldStartPreviewController: boolean): void => {
  topBarEventEmitter.emit('SET_SHOULD_START_PREVIEW_CONTROLLER', shouldStartPreviewController);
};

const setStartPreviewCallback = (callback: () => void): void => {
  topBarEventEmitter.emit('SET_START_PREVIEW_CALLBACK', callback);
};

export default {
  updateTopBar,
  setElement,
  setFileName,
  setHasUnsavedChange,
  getTopBarPreviewMode,
  setShouldStartPreviewController,
  setStartPreviewCallback,
};
