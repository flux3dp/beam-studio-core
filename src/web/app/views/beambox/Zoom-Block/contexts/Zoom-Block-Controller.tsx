import { ZoomBlockContextHelper } from 'app/views/beambox/Zoom-Block/Zoom-Block';

const updateZoomBlock = () => {
  if (!ZoomBlockContextHelper.context) {
    console.log('ZoomBlock is not mounted now.');
  } else {
    ZoomBlockContextHelper.context.updateZoomBlock();
  }
};

export default {
  updateZoomBlock,
};
