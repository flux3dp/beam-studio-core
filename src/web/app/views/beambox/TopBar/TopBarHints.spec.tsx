/* eslint-disable import/first */
import * as React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

jest.mock('helpers/i18n', () => ({
  lang: {
    topbar: {
      hint: {
        polygon: 'Press + / - key to increase / decrease sides.',
      },
    },
  },
}));

import { TopBarHintsContext } from 'app/views/beambox/TopBar/contexts/TopBarHintsContext';

import TopBarHints from './TopBarHints';

test('should render correctly', () => {
  expect(toJson(mount(
    <TopBarHintsContext.Provider value={{
      hintType: null,
    }}
    >
      <TopBarHints />
    </TopBarHintsContext.Provider>,
  ))).toMatchSnapshot();

  expect(toJson(mount(
    <TopBarHintsContext.Provider value={{
      hintType: 'POLYGON',
    }}
    >
      <TopBarHints />
    </TopBarHintsContext.Provider>,
  ))).toMatchSnapshot();

  expect(toJson(mount(
    <TopBarHintsContext.Provider value={{
      hintType: 'CIRCLE',
    }}
    >
      <TopBarHints />
    </TopBarHintsContext.Provider>,
  ))).toMatchSnapshot();
});
