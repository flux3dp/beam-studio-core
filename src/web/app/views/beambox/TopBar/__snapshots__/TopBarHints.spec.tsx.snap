// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<TopBarHints>
  <div
    className="hint-container"
  />
</TopBarHints>
`;

exports[`should render correctly 2`] = `
<TopBarHints>
  <div
    className="hint-container"
  >
    <div>
      Press + / - key to increase / decrease sides.
    </div>
  </div>
</TopBarHints>
`;

exports[`should render correctly 3`] = `
<TopBarHints>
  <div
    className="hint-container"
  />
</TopBarHints>
`;
